x11-libs/libdlo: New package

ConiKost opened the pull request:
x11-libs/libdlo: New package

Closes: https://bugs.gentoo.org/652702

a17r commented on x11-libs/libdlo/metadata.xml:
```diff
@@ -0,0 +1,23 @@
+		It's still very much under development, but it's possible to write custom applications
+		to it today (or to libdlo user mode library itself, bypassing udlfb).
+		And it's also possible to configure X and standard X applications to run on udlfb
+		but that part is still very distro dependent today. Most people in the DisplayLink
```
That description seems to be overly long, even if it is called `longdescription`. At least the last two sentences seem superfluous. And what about the kernel number reference relative to 'to date' - it could be a distraction or make it appear this being a very old (?) package contrary to being 'still very much under development'.

ConiKost replied:
I've changed the description, which shoud now sound better.

ConiKost commented:
> Package installs /usr/bin/test1
From what I understand, this tool can be used to test you own displaylink device. I've made an use flag "test-program" for this, on which the installation depends and renamed it to displaylink-test.

> Unnecessary /usr/lib64/libdlo.la should be removed via the usual find "${D}" -name '*.la' -delete || die in src_install.

Added to the eBuild.

> Don't install /usr/lib64/libdlo.a unless USE=static-libs was set.
Fixed, it depends now on IUSE="static-libs".

a17r commented on x11-libs/libdlo/libdlo-0.1.2.ebuild:
```diff
@@ -0,0 +1,55 @@
+	# archiver requires 'AM_PROG_AR' in 'configure.ac'.
+	sed -i -e '/AC_PROG_CC/a AM_PROG_AR' configure.ac || die
+
+	# Only build the Displaylink test program, if a user want's it.
```
no apostrophe in `wants`, same on next line ;)

ConiKost replied:
Whoops ;-) Fixed.

ConiKost commented on x11-libs/libdlo/libdlo-0.1.2.ebuild:
```diff
@@ -0,0 +1,55 @@
+		sed -i -e '/test\/images.*/d' Makefile.am || die
+		sed -i -e '/TESTS .*/d' Makefile.am || die
+	else
+		sed -i -e 's/TESTS .*/TESTS = test\/displaylink-test/' Makefile.am || die
```
Fine for me. Just did it this way, as someone told me, that I should avoid using things like mv in src_install and patch the build system.

a17r replied:
ah, that depends, but this case is trivial. again, a build system patch that you can take upstream is always preferred, so that these workarounds can at some point be dropped.

ConiKost replied:
Okay, changed now in newest version.

ConiKost commented on x11-libs/libdlo/libdlo-0.1.2.ebuild:
```diff
@@ -0,0 +1,55 @@
+	# Only build the Displaylink test program, if a user want's it.
+	# If a user want's it, rename test1 to displaylink-test.
+	if ! use test-program; then
+		sed -i -e '/test\/Makefile/d' configure.ac || die
```
I've done a patch now :)
